public/app/plugins/panel/geomap/layers/data/heatMap.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  FieldCalcs,
+  FieldType,
+  getFieldColorModeForField,
+  GrafanaTheme2,
+  MapLayerHandler,
+  MapLayerOptions,
+  MapLayerRegistryItem,
+  PanelData,
+  reduceField,
+  ReducerID,
+} from '@grafana/data';
+import Map from 'ol/Map';
+import Feature from 'ol/Feature';
+import * as layer from 'ol/layer';
+import * as source from 'ol/source';
+import { dataFrameToPoints, getLocationMatchers } from '../../utils/location';
+
+// Configuration options for Heatmap overlays
+export interface HeatmapConfig {
+  blur: number;
+  radius: number;
+}
+
+const defaultOptions: HeatmapConfig = {
+  blur: 15,
+  radius: 5,
+};
+
+/**
+ * Map layer configuration for heatmap overlay
+ */
+export const heatmapLayer: MapLayerRegistryItem<HeatmapConfig> = {
+  id: 'heatmap',
+  name: 'Heatmap',
+  description: 'visualizes a heatmap of the data',
+  isBaseMap: false,
+  showLocation: true,
+
+  /**
+   * Function that configures transformation and returns a transformer
+   * @param options
+   */
+  create: (map: Map, options: MapLayerOptions<HeatmapConfig>, theme: GrafanaTheme2): MapLayerHandler => {
+    const config = { ...defaultOptions, ...options.config };
+    const matchers = getLocationMatchers(options.location);
+
+    const vectorSource = new source.Vector();
+
+    // Create a new Heatmap layer
+    // Weight function takes a feature as attribute and returns a normalized weight value
+    const vectorLayer = new layer.Heatmap({
+      source: vectorSource,
+      blur: config.blur,
+      radius: config.radius,
+      weight: function (feature) {
+        var weight = feature.get('value');
+        return weight;
+      },
+    });
+
+    return {
+      init: () => vectorLayer,
+      update: (data: PanelData) => {
+        const frame = data.series[0];
+
+        // Remove previous data before updating
+        const features = vectorLayer.getSource().getFeatures();
+        features.forEach((feature) => {
+          vectorLayer.getSource().removeFeature(feature);
+        });
+
+        // Get data points (latitude and longitude coordinates)
+        const info = dataFrameToPoints(frame, matchers);
+        if(info.warning) {
+          console.log( 'WARN', info.warning);
+          return; // ???
+        }
+
+        // Get the field of data values
+        const field = frame.fields.find(field => field.type === FieldType.number); // TODO!!!!
```

Comment:
We should make this configurable, and default to "count" -- for configuration, lets use the work starting in:
https://github.com/grafana/grafana/pull/36825

so I will go ahead and merge this.

Great work @eunice98k